core/ui/src/lib/components/repo/ModifiedFilesCard.svelte

```diff
@@ -120,10 +120,11 @@
 	const cleanUpChangeSets = async () => {
 		ensureDefaultChangeset();
 		for (let i = 0; i < changeSets.length; i += 1) {
-			// Keep only files that exist in modifiedFiles
-			changeSets[i].files = changeSets[i].files.filter((file) =>
-				modifiedFiles.some((mf) => mf.path === file.path)
-			);
+			// Keep only files that exist in modifiedFiles, while updating the file state to match the entry in modifiedFiles
+			changeSets[i].files = changeSets[i].files.reduce<ModifiedFile[]>((acc, file) => {
+				const updatedFile = modifiedFiles.find((mf) => mf.path === file.path);
+				return updatedFile ? [...acc, updatedFile] : acc;
+			}, []);
```

Comment:
`reduce` syntax can be confusing to look at, i have to read the docs every time. basically it uses an accumulator `acc`, and passes that into each run through the loop. `acc` is a single value, in this case a new array of files. if we find a file in `modifiedFiles`, it's allowed to stay in the changeset, so we push it into the accumulator. if it's not in modifiedFiles, it's not allowed to stay in the changeSet, so we do nothing to `acc`. 

tldr, this uses `reduce` to both:
1. filter out anything in the changeSet that isn't in modifiedFiles anymore, and
2. update file info (like lock status) for each file that DOES exist in modifiedFiles

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce